src/blocks.c

```diff
@@ -468,7 +468,6 @@ static void process_footnotes(cmark_parser *parser) {
   while ((ev_type = cmark_iter_next(iter)) != CMARK_EVENT_DONE) {
     cur = cmark_iter_get_node(iter);
     if (ev_type == CMARK_EVENT_EXIT && cur->type == CMARK_NODE_FOOTNOTE_DEFINITION) {
-      cmark_node_unlink(cur);
```

Comment:
The changes in this file support [allowing nested foornotes](https://github.com/github/cmark-gfm/pull/229/commits/71e27f25f11c9a34f0532dba459940e1fb5f6316).

We accomplish this by simply changing _when_ we unlink an unused footnote node.

Whereas before, we iterate thru the parser and delete all footnote definitions as we see them

Now we only unlink the footnotes definitions _after we've scanned them for more references_.